classes/salesforce_push.php

```diff
@@ -433,21 +442,30 @@ function check_user( $push_allowed, $object_type, $object, $sf_sync_trigger, $ma
 					// skip this object if it is a draft and the fieldmap settings told us to ignore it
 					continue;
 				}
+
 				if ( isset( $mapping['push_async'] ) && ( '1' === $mapping['push_async'] ) && false === $manual ) {
 					// this item is async and we want to save it to the queue
-					$data = array(
+
+					// if we determine that the below code does not perform well, worst case scenario is we could save $data to a custom table, and pass the id to the callback method.
+					/*$data = array(
 						'object_type'     => $object_type,
 						'object'          => $object,
-						'mapping'         => $mapping,
+						'mapping'         => $mapping['id'],
 						'sf_sync_trigger' => $sf_sync_trigger,
+					);*/
+
+					// add a queue action to push data to salesforce
+					// this means we don't need the frequency for this method anymore, i think
```

Comment:
Agreed. From what I see, adding it to the queue will get the async push processed eventually, without any frequency attached.